sys/include/net/gnrc/mac/internal.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_mac  A common MAC type for providing key MAC parameters and helper functions
+ * @ingroup     net
+ * @brief       A common MAC type for providing key MAC parameters and helper functions.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for internal functions of MAC protocol
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_MAC_INTERNAL_H_
+#define GNRC_MAC_INTERNAL_H_
+
+#include <stdint.h>
+#include <types.h>
+#include <net/ieee802154.h>
+#include <net/gnrc/netdev2.h>
+#include <net/gnrc/mac/mac.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if GNRC_MAC_NEIGHBOUR_COUNT != 0
+/* @brief fetch the address of the tx neighbor according to the neighbor's ID
+ *
+ * @param[in]     tx        internal state of transmission state machine
+ * @param[in]     id        the ID of the neighbor
+ *
+ * @return                  the address of the neighbor
+ */
+static inline gnrc_mac_tx_neighbour_t* _get_neighbour(gnrc_mac_tx_t* tx, unsigned int id)
+{
+    return &(tx.neighbours[id]);
+}
+#endif
+
+#if GNRC_MAC_TX_QUEUE_SIZE != 0
+/* @brief queue the packet into the related transmission packet queue.
+ *        Note that, in case the neighbor structure is used and the neighbor has Tx-queue,
+ *        this function queues the packet to the queue associated with the
+ *        pkt's destination neighbor. On the other hand, if neighbor structure is not used,
+ *        this function queues the packet to the single priority TX queue stored in
+ *        the 'gnrc_mac_tx_t' structure.
```

Comment:
Has adapted `_queue_tx_packet()` function, trying to be more general for queueing Tx packets.  

- case-1: When `neighbour` structure is not adopted, `_queue_tx_packet()` queues the packet to the single Tx priority queue in `netdev2->tx`.

- case-2: When `neighbour` structure is adopted and includes packet queue inside it, `_queue_tx_packet()` queues the packet to the neighbour's specific Tx queue.